iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -488,6 +490,9 @@ private void commitWrite(Table table, String branchName, Long snapshotId, long s
       if (snapshotId != null) {
         write.validateFromSnapshot(snapshotId);
       }
+      Expression conflictDetectionFilter =  Expressions.and(Expressions.alwaysTrue(), filterExpr);
```

Comment:
why is this AND true required, isn't
True And expr = filterExpr?
The filterExpr was build with these AND only right?

Reply:
Incase there are no filters in the query, we will have to pass true to the conflict detection filter. For this case the Expressions.alwaysTrue(), filterExpr) is done.